Controllers/LoginController.cs

```diff
@@ -39,27 +39,30 @@ public Models.Login Get() => _tools.IsLogin(out var displayName)
 			return new() { Success = false, Code = 7, Message = "尝试登录次数过多，请在30分钟后重试。" };
 		}
 		if (account.Length is < 4 or > 32 || !ICheckingTools.IsPasswordComplicated(password)) {
-			_ = Hubs.Cache.Set($"TryLoginCount of {account}", count++, TimeSpan.FromMinutes(30), TimeSpan.FromHours(2));
+			_ = Hubs.Cache.Set($"TryLoginCount of {account}", ++count, TimeSpan.FromMinutes(30), TimeSpan.FromHours(2));
 			return new() { Success = false, Code = 6, Message = "用户名或密码错误。" };
 		}
-		using var reader = _provider.GetUserReader(account);
+		using var reader = _provider.GetUserReader(account, out var cmd);
 		if (!reader.Read()) {
-			_ = Hubs.Cache.Set($"TryLoginCount of {account}", count++, TimeSpan.FromMinutes(30), TimeSpan.FromHours(2));
+			_ = Hubs.Cache.Set($"TryLoginCount of {account}", ++count, TimeSpan.FromMinutes(30), TimeSpan.FromHours(2));
 			return new() { Success = false, Code = 6, Message = "用户名或密码错误。" };
 		}
 		var hash = new byte[64];
 		_ = reader.GetBytes(3, 0, hash, 0, 64);
 		var salt = new byte[16];
 		_ = reader.GetBytes(4, 0, salt, 0, 16);
 		if (!ICheckingTools.VerifyPassword(password, hash, salt)) {
-			_ = Hubs.Cache.Set($"TryLoginCount of {account}", count++, TimeSpan.FromMinutes(30), TimeSpan.FromHours(2));
+			_ = Hubs.Cache.Set($"TryLoginCount of {account}", ++count, TimeSpan.FromMinutes(30), TimeSpan.FromHours(2));
 			return new() { Success = false, Code = 6, Message = "用户名或密码错误。" };
 		}
 		Hubs.Cache.MemoryCache.Remove($"TryLoginCount of {account}");
-		Session.SetString("Name", account);
-		Session.SetString("Nick", reader.GetString(2));
-		Session.Set("Hash", hash);
-		Session.Set("Salt", salt);
-		return new() { Success = true, Code = 0, Message = "登录成功！" };
+		HttpContext.Session.SetString("Name", account);
+		HttpContext.Session.SetString("Nick", reader.GetString(2));
+		HttpContext.Session.Set("Hash", hash);
+		HttpContext.Session.Set("Salt", salt);
+		cmd.Dispose();
+		_logger.LogDebug("Login: {}登陆成功!", account);

```

Comment:
引入 IConnectionInfo，然后
```suggestion
		_logger.LogDebug("Login: 用户 {} 于 {} 登录成功。", account, address);

```

Reply:
尝试登录次数过多干啥不加一个 LogInfomation？ @execute233